pkg/controller/servicemesh/controlplane/reconciler.go

```diff
@@ -490,13 +502,30 @@ func (r *ControlPlaneReconciler) postReconciliationStatus(reconciliationReason v
 	if processingErr == nil {
 		reconciledCondition.Message = reconciliationMessage
 	} else {
-		reconciledCondition.Message = fmt.Sprintf("%s: error: %s", reconciliationMessage, processingErr)
-		r.Manager.GetRecorder(controllerName).Event(r.Instance, corev1.EventTypeWarning, reason, reconciliationMessage)
+		// grab the cause, as it's likely the error includes the reconciliation message
+		reconciledCondition.Message = fmt.Sprintf("%s: error: %s", reconciliationMessage, errors.Cause(processingErr))
+		r.Manager.GetRecorder(controllerName).Event(r.Instance, corev1.EventTypeWarning, reason, reconciledCondition.Message)
 	}
 	r.Status.SetCondition(reconciledCondition)
+
+	// we should only post status updates if condition status as changed
+	if r.skipStatusUpdate() {
+		return nil
+	}
+
 	return r.PostStatus()
 }
 
+func (r *ControlPlaneReconciler) skipStatusUpdate() bool {
+	for _, conditionType := range []v1.ConditionType{v1.ConditionTypeInstalled, v1.ConditionTypeReconciled, v1.ConditionTypeReady} {
+		// TODO: what if the status is the same, but the reason is different? - we should also compare the reason
```

Comment:
I think only checking the status is correct. The reason only applies when the status actually transitions from one state to the other. We should never update the status if the state doesn't change, regardless if the underlying reason changes.

Reply:
removed